packages/client/README.md

```diff
@@ -114,257 +205,280 @@ const msg = {
+    {
+        last: 10,
+    }, 
+    messageCallback
+)
+```
 
+Or fetch them and subscribe to new messages in the same call via a `subscribe` call:
 ```js
-const client = new StreamrClient({
-    auth: {
-        privateKey: 'your-private-key'
+// Fetches the last 10 messages and subscribes to the stream
+const sub1 = await streamr.subscribe({
+    id: streamId,
+    resend: {
+        last: 10,
     }
-})
+}, messageCallback)
 ```
 
-Authenticating with an Ethereum private key contained in an Ethereum (web3) provider:
-
+Resend from a specific message reference up to the newest message:
 ```js
-const client = new StreamrClient({
-    auth: {
-        ethereum: window.ethereum,
+const sub2 = await streamr.subscribe({
+    id: streamId,
+    resend: {
+        from: {
+            timestamp: (Date.now() - 1000 * 60 * 5), // 5 minutes ago
+        },
+        publisher: '0x12345...', // optional
     }
-})
+}, onMessage)
 ```
-
-Authenticating with a pre-existing session token (used internally by the Streamr app):
-
+Resend a limited range of messages:
 ```js
-const client = new StreamrClient({
-    auth: {
-        sessionToken: 'session-token'
+const sub3 = await streamr.subscribe({
+    id: streamId,
+    resend: {
+        from: {
+            timestamp: (Date.now() - 1000 * 60 * 10), // 10 minutes ago
+        },
+        to: {
+            timestamp: (Date.now() - 1000 * 60 * 5), // 5 minutes ago
+        },
+        // when using from and to the following parameters are optional
+        // but, if specified, both must be present
+        publisher: '0x12345...', 
+        msgChainId: 'ihuzetvg0c88ydd82z5o', 
     }
-})
+}, onMessage)
 ```
-
-To extract the session token from an authenticated client:
+If you choose one of the above resend options when subscribing, you can listen on the completion of this resend by doing the following:
 
 ```js
-const bearerToken = await client.session.getSessionToken()
+const sub = await streamr.subscribe(options)
+sub.onResent(() => {
+    console.log('All caught up and received all requested historical messages! Now switching to real time!')
+})
 ```
 
-Then for example,
+### Search Streams
+You can search for a stream which has the `term` string in it's id as follows:
 ```js
-    axios({
-        headers: {
-            Authorization: `Bearer ${bearerToken}`,
-        },
-        ...
-    )}
+const streams = await streamr.searchStreams('foo')
 ```
+You can query for the streams using an optional second parameter to fine-tune your search. A permission query searches over stream permissions. You can either query by direct permissions (which are explicitly granted to a user), or by all permissions (including public permissions, which apply to all users).
 
-Note, session tokens expire after four hours and may need to be refreshed.
-
-## Connecting
+To get all streams where a user has some direct permission. The `user` option can be omitted. In that case, it defaults to the authenticated user:
```

Comment:
It doesn't default to the authenticated user. The cli-tools command does, when we define a `--user` flag.